src/systemctl/systemctl-show.c

```diff
@@ -433,6 +433,13 @@ static void print_status_info(
                                FORMAT_TIMESTAMP_STYLE(until_timestamp, arg_timestamp_style),
                                FORMAT_TIMESTAMP_RELATIVE(until_timestamp));
                 }
+
+                if (STRPTR_IN_SET(i->active_state, "inactive", "failed")) {
+                        usec_t duration;
+
+                        duration = i->active_exit_timestamp - i->active_enter_timestamp;
```

Comment:
maybe as safety check also in the conditionalization of this field, that both timestamps are set and the exit timestamp actually larger than the enter timestamp, and if not suppress the output of this entry.

Reply:
hmm, so you are using the 'active' timestamps here. i.e. you cover the time the service was fully up. Type=oneshot services will never be "active" though, they only have an "activating" phase, and once that is complete they are immediately dead again. Question is, what to do about those. Two options:

1. maybe use the non-inactive time. In that case we'd include ExecStartPre=/ExecStart= and ExecStop= times and stuff in the "duration". And the Type=oneshot services would get useful output too.

2. accept that this is the way it is and ensure that we suppress output in the type=inactive case. This should not require extra work, if the above safety checks are added, as for Type=oneshot the active-enter timestamp never is set after all..

I am kind leaning towards option 2, but I figure people will be surprised. But I'd accept it.

Reply:
Went for the second option and added the necessary checks.